feat(autoware_lanelet_map_validator): add dangling reference checker to non existing intersection_area (#177)
autowarefoundation/autoware_tools: map/autoware_lanelet2_map_validator/docs/intersection/intersection_area_dangling_reference.md

# intersection_area_dangling_reference

## Validator name

mapping.intersection.intersection_area_dangling_reference

## Feature

> This validator check whether each intersection lanelet(namely the lanelet with `turn_direction` property) has existing reference to `intersection_area` polygon. The countercase occurs when an existing intersection_area is deleted but its referrers are not updated.

Author: ```suggestion
This validator checks whether each intersection lanelet(namely the lanelet with `turn_direction` property) has reference to existing `intersection_area` polygon. Such invalid case occurs when an intersection_area is deleted but its referrers are not updated.
```

This is achieved by the following procedure.

1. Obtain the set of `intersection_area` polygon IDs
2. Check if intersection lanelet has "intersection_area" key and its value is contained in the above IDs

The validator outputs the following issue with the corresponding ID of the primitive.

| Issue Code                                         | Message                                                                                 | Severity | Primitive | Description                                                                        | Approach                                                             |
| -------------------------------------------------- | --------------------------------------------------------------------------------------- | -------- | --------- | ---------------------------------------------------------------------------------- | -------------------------------------------------------------------- |

> | Intersection.IntersectionAreaDanglingReference-001 | "Dangling reference to non-existing intersection area of ID \<LANELET ID\> is detected" | Error    | Lanelet   | Lookup to `intersection_area` from the reporeted lanelet will cause runtime error. | Go to the reported lanelet and delete "intersection_area" key entry. |

Author: ```suggestion
| Intersection.IntersectionAreaDanglingReference-001 | "Dangling reference to non-existing intersection area of ID \<LANELET ID\> is detected" | Error    | Lanelet   | The reported lanelet will cause a runtime error when attempting to look up the non-existent intersection_area. | Go to the reported lanelet and delete "intersection_area" key entry. |
```
